pkg/scheduler/apis/config/types_pluginargs.go

```diff
@@ -86,6 +86,30 @@ type RequestedToCapacityRatioArgs struct {
 	Resources []ResourceSpec
 }
 
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// NodeResourcesLeastAllocatedArgs holds arguments used to configure NodeResourcesLeastAllocated plugin.
+type NodeResourcesLeastAllocatedArgs struct {
+	metav1.TypeMeta
+
+	// Resources to be managed, if no resource is provided, default resource set with both
+	// the weight of "cpu" and "memory" set to "1" will be applied.
+	// Resource with "0" weight will not accountable for the final score.
+	Resources []ResourceSpec
```

Comment:
now that ResourceSpec is used by multiple args, update the type/field doc to remove references to RequestedToCapacityRatio

Reply:
are the names/weights validated? what are valid values for weight?

Reply:
Added the logic to validate the weight in the plugins, pls see my response later.  In the long term, I think we might need something like `jsonschema validator` to validate the yaml configuration, this is currently not implemented in the project, I will take a investigation first.

Reply:
Are you referring to this? #90859